lightly/api/api_workflow_datasets.py

```diff
@@ -0,0 +1,96 @@
+
+        Args:
+            dataset_name:
+                The name of the dataset for which the dataset_id
+                should be set as attribute
+
+        Returns:
+
+        Raises: ValueError
+
+        '''
+        current_datasets: List[DatasetData] \
+            = self.datasets_api.get_datasets()
+
+        try:
+            dataset_with_specified_name = next(dataset for dataset in current_datasets if dataset.name == dataset_name)
+            self._dataset_id = dataset_with_specified_name.id
+        except StopIteration:
+            raise ValueError(f"A dataset with the name {dataset_name} does not exist on the web platform. "
+                             f"Please create it first.")
+
+    def create_dataset(self: ApiWorkflowClient, dataset_name: str):
+        '''Creates a dataset on the webplatform
+
+        If a dataset with that name already exists, instead the dataset_id is set.
+        Args:
+            dataset_name:
+                The name of the dataset to be created.
+
+        '''
+        try:
+            self.set_dataset_id(dataset_name)
+        except ValueError:
+            self._create_dataset_without_check_existing(dataset_name=dataset_name)
+
+    def _create_dataset_without_check_existing(self: ApiWorkflowClient, dataset_name: str):
+        ''' Creates a dataset on the webplatform
+
+        No checking if a dataset with such a name already exists is performed.
+        Args:
+            dataset_name:
+                The name of the dataset to be created.
+
+        '''
+        body = DatasetCreateRequest(name=dataset_name)
+        response: CreateEntityResponse = self.datasets_api.create_dataset(body=body)
+        self._dataset_id = response.id
+
+    def create_new_dataset_with_counter(self, dataset_basename: str):
+        '''Creates a new dataset on the web platform
+
+        If a dataset with the specified name already exists,
+        a counter is added to the name to be able to still create it.
+        Args:
+            dataset_basename:
+                The name of the dataset to be created.
+
+        '''
+        current_datasets: List[DatasetData] \
+            = self.datasets_api.get_datasets()
+        current_datasets_names = [dataset.name for dataset in current_datasets]
+
+        if dataset_basename not in current_datasets_names:
+            self._create_dataset_without_check_existing(dataset_name=dataset_basename)
+            return
+
+        counter = 1
+        dataset_name = f"{dataset_basename}_{counter}"
+        while dataset_name in current_datasets_names:
+            counter += 1
+            dataset_name = f"{dataset_basename}_{counter}"
+        self._create_dataset_without_check_existing(dataset_name=dataset_name)
+
+    def delete_dataset(self: ApiWorkflowClient):
+        ''' Deletes a dataset on the web platform
+
+        '''
+        self.datasets_api.delete_dataset_by_id(dataset_id=self.dataset_id)
```

Comment:
This function is dangerous! It takes the default dataset_id and just deletes it? I see big problems here...

Please remove this one and replace it by 'delete_dataset_by_id'. This way the user has at least to provide the id to delete a dataset.